crates/env/src/call/execution.rs

```diff
@@ -97,12 +96,30 @@ impl<Abi> ExecutionInput<EmptyArgumentList<Abi>, Abi> {
     #[inline]
     pub fn new(selector: Selector) -> Self {
         Self {
-            selector,
+            selector: Some(selector),
             args: ArgumentList::empty(),
             _marker: Default::default(),
         }
     }
+}
 
+impl ExecutionInput<EmptyArgumentList<Sol>, Sol> {
+    /// Creates a new execution input with no selector.
+    ///
+    /// # Note
+    ///
+    /// Should only be used for Solidity ABI encoded constructors/instantiation.
+    #[inline]
+    pub fn no_selector() -> Self {
```

Comment:
Happy to see that you implemented this here, we'll also need it for calling the pre-compiles.